### PR TITLE
[BAEL-3219] - Parsing an XML File Using SAX Parser

### PR DESCRIPTION
**Pull request by YassinHajaj**
Parsing an XML File Using SAX Parser

### CODE REVIEW

File: xml/src/main/java/com/baeldung/sax/SaxParserMain.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+import java.util.List;
+import java.util.Objects;
+
+public class SaxParserMain {
+    public static void main(String[] args) throws ParserConfigurationException, SAXException, IOException {
+        SAXParserFactory factory = SAXParserFactory.newInstance();
+        SAXParser saxParser = factory.newSAXParser();
+
+        BaeldungHandler baeldungHandler = new BaeldungHandler();
+        saxParser.parse("xml\\src\\main\\resources\\sax\\baeldung.xml", baeldungHandler);
```
Comment (rpvilao) [inline xml/src/main/java/com/baeldung/sax/SaxParserMain.java]: can we use forward slash here?

File: xml/src/main/java/com/baeldung/sax/SaxParserMain.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+        @Override
+        public String toString() {
+            return "BaeldungHandler{" +
+                    "website=" + website +
+                    ", elementValue='" + elementValue + '\'' +
+                    '}';
+        }
+
+        private static final String ARTICLES = "articles";
```
Comment (rpvilao) [inline xml/src/main/java/com/baeldung/sax/SaxParserMain.java]: move to the top

File: xml/src/main/java/com/baeldung/sax/SaxParserMain.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+            return articleList;
+        }
+
+        public void setArticleList(List<BaeldungArticle> articleList) {
+            this.articleList = articleList;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
```
Comment (rpvilao) [inline xml/src/main/java/com/baeldung/sax/SaxParserMain.java]: brackets on all if, for, while etc

File: xml/src/main/java/com/baeldung/sax/SaxParserMain.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+            if (o == null || getClass() != o.getClass()) return false;
+            Baeldung baeldung = (Baeldung) o;
+            return Objects.equals(articleList, baeldung.articleList);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(articleList);
```
Comment (rpvilao) [inline xml/src/main/java/com/baeldung/sax/SaxParserMain.java]: Consider using lombok for those

File: xml/src/main/java/com/baeldung/sax/SaxParserMain.java [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            BaeldungArticle that = (BaeldungArticle) o;
+            return Objects.equals(title, that.title) &&
+                    Objects.equals(content, that.content);
+        }
+
+        @Override
```
Comment (rpvilao) [inline xml/src/main/java/com/baeldung/sax/SaxParserMain.java]: same

File: xml/src/main/java/com/baeldung/sax/SaxParserMain.java [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+        public String getTitle() {
+            return title;
+        }
+
+        public void setTitle(String title) {
+            this.title = title;
+        }
+
+        public String getContent() {
```
Comment (rpvilao) [inline xml/src/main/java/com/baeldung/sax/SaxParserMain.java]: lombok setters and getters

File: xml/src/main/java/com/baeldung/sax/SaxParserMain.java (hunk 1 above)
Comment (YassinHajaj) [reply xml/src/main/java/com/baeldung/sax/SaxParserMain.java]: Yes, I corrected it in the test but not here. Good point

**Pull request merged**